test/unit/tools/test_metadata.py

```diff
@@ -0,0 +1,171 @@
+        with open(output_dataset.dataset.file_name, "w") as f:
+            f.write(contents)
+
+    def _write_galaxy_json(self, contents):
+        job_metadata = os.path.join(self.tool_working_directory, self.tool.provided_metadata_file)
+        with open(job_metadata, "w") as f:
+            f.write(contents)
+
+    def metadata_command(self, output_datasets):
+        metadata_compute_strategy = get_metadata_compute_strategy(self.app, self.job.id)
+        self.metadata_compute_strategy = metadata_compute_strategy
+
+        exec_dir = None
+        dataset_files_path = self.app.model.Dataset.file_path
+        config_root = self.app.config.root
+        config_file = None
+        datatypes_config = os.path.join(self.job_working_directory, 'registry.xml')
+        self.app.datatypes_registry.to_xml_file(path=datatypes_config)
+        job_metadata = os.path.join(self.tool_working_directory, self.tool.provided_metadata_file)
+        output_fnames = [DatasetPath(o.dataset.id, o.dataset.file_name, None) for o in output_datasets.values()]
+        command = metadata_compute_strategy.setup_external_metadata(output_datasets,
+                                                                    self.app.model.session,
+                                                                    exec_dir=exec_dir,
+                                                                    tmp_dir=self.job_working_directory,  # set in jobs/runners.py - better if was default.
+                                                                    dataset_files_path=dataset_files_path,
+                                                                    config_root=config_root,
+                                                                    config_file=config_file,
+                                                                    datatypes_config=datatypes_config,
+                                                                    job_metadata=job_metadata,
+                                                                    output_fnames=output_fnames,
+                                                                    max_metadata_value_size=10000)
+        return command
+
+    def exec_metadata_command(self, command):
+        with open(self.stdout_path, "wb") as stdout_file, open(self.stderr_path, "wb") as stderr_file:
+            _environ = os.environ.copy()
+            _environ["PYTHONPATH"] = os.path.abspath("lib")
+            proc = subprocess.Popen(args=command,
+                                    shell=True,
+                                    cwd=self.job_working_directory,
+                                    env=_environ,
+                                    stdout=stdout_file,
+                                    stderr=stderr_file)
+            ret = proc.wait()
+        self.print_command_output()
+        assert ret == 0
+        return ret
+
+    def print_command_output(self):
```

Comment:
If this should stay, maybe some more debug lines are useful to know what is actually printed.

Reply:
Really good point, I'll rebase this with some context!